lib/ts/nextjs/middleware.ts

```diff
@@ -0,0 +1,204 @@
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            credentials: "include",
+        });
+
+        if (signOutResponse.status !== 200) {
+            return;
+        }
+
+        const response = new Response(null, {});
+        response.headers.set("x-middleware-next", "1");
+        // TODO: Delete only the cookies that are auth related
+        response.headers.delete("set-cookie");
+        response.headers.delete(ACCESS_TOKEN_HEADER_NAME);
+        response.headers.delete(REFRESH_TOKEN_HEADER_NAME);
+        response.headers.delete(FRONT_TOKEN_HEADER_NAME);
+        response.headers.delete(ANTI_CSRF_TOKEN_HEADER_NAME);
+        return response;
+    } catch (err) {
+        logDebugMessage("Error revoking session");
+        logDebugMessage(err as unknown as string);
+        return;
+    }
+}
+
+function redirectToAuthPage(request: Request): Response {
+    const authPagePath = AppInfo.websiteBasePath || "/auth";
+    const redirectUrl = new URL(authPagePath, request.url);
+    redirectUrl.searchParams.set(FORCE_LOGOUT_PATH_PARAM_NAME, "true");
+    logDebugMessage(`Redirecting to: ${redirectUrl}`);
+    return new Response(null, {
+        status: 307,
+        headers: {
+            Location: redirectUrl.toString(),
+        },
+    });
+}
+
+async function fetchNewTokens(currentRefreshToken: string): Promise<{
+    accessToken: SuperTokensRequestToken;
+    refreshToken: SuperTokensRequestToken;
+    frontToken: SuperTokensRequestToken;
+    antiCsrfToken: SuperTokensRequestToken;
+}> {
+    const refreshApiURL = new URL(`${AppInfo.apiBasePath}/session/refresh`, AppInfo.apiDomain);
+    const refreshResponse = await fetch(refreshApiURL, {
+        method: "POST",
+        headers: {
+            "Content-Type": "application/json",
+            Cookie: `sRefreshToken=${currentRefreshToken}`,
+        },
+        credentials: "include",
+    });
+    logDebugMessage("Session refresh request completed");
+    const frontToken: SuperTokensRequestToken = {
+        header: refreshResponse.headers.get("front-token"),
+        cookie: `${FRONT_TOKEN_COOKIE_NAME}=${refreshResponse.headers.get("front-token")}; Path=/`,
+    };
+    const accessToken: SuperTokensRequestToken = {
+        header: refreshResponse.headers.get(ACCESS_TOKEN_HEADER_NAME),
+        cookie: null,
+    };
+    const refreshToken: SuperTokensRequestToken = {
+        header: refreshResponse.headers.get(REFRESH_TOKEN_HEADER_NAME),
+        cookie: null,
+    };
+    const antiCsrfToken: SuperTokensRequestToken = {
+        header: refreshResponse.headers.get(ANTI_CSRF_TOKEN_HEADER_NAME),
+        cookie: null,
+    };
+
+    // getSetCookie was added in node 18 and our build target is ES5
+    // This should not a problem here since the function runs in the Vercel edge runtime environment
+    // @ts-expect-error TS(2339): Property 'getSetCookie' does not exist on type 'Headers'.
```

Comment:
As a TODO: let's review if we could raise our build-target

Reply:
resolved